tests/plotting/conftest.py

```diff
@@ -20,10 +20,17 @@ def _is_vtk(obj):
 
 
 @pytest.fixture(autouse=True)
-def check_gc():
+def check_gc(request):
     """Ensure that all VTK objects are garbage-collected by Python."""
     before = set(id(o) for o in gc.get_objects() if _is_vtk(o))
     yield
+
+    # Do not check for collection if the test failed. Tests that fail also fail
```

Comment:
When you say "when the test failed", do you mean "when the test suite failed", i.e. "there was at least one failing test"? Because that's what this seems to do (and I agree with that). Doing it on a per-test basis would be a bit messier, according to https://doc.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures and https://github.com/pytest-dev/pytest/issues/230.

Reply:
You're correct, and I've updated the comment in f76f8f05fd87a400f6b5f03df9f02a7af3df2df6.